### PR TITLE
feat: Added search input text font family option

### PR DESCRIPTION
**Pull request by summerkiflain**
I am using custom font in my whole app, so I added support for custom fontFamily in this PR.
- For now I am using patch-package to monkey patch the latest version, this PR will help me remove that dependency.

### CODE REVIEW

**Comment by summerkiflain** [comment]
Hi @eriveltonelias, please review, I have added another commit, all checks are passing now, and please please publish on npm after merging this.

File: src/SearchLayout.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default class SearchLayout extends React.Component {
             onSubmit={this._handleSubmit}
             placeholderTextColor={this.props.searchInputPlaceholderTextColor}
             textColor={this.props.searchInputTextColor}
-            textFont={this.props.searchInputTextFont}
+            textFontFamily={this.props.searchInputTextFontFamily}
```
Comment (eriveltonelias) [inline src/SearchLayout.js:46]: could you add this new prop to the types definition file too?

File: src/SearchBar.ios.js [hunk 2]
```diff
@@ -154,10 +154,14 @@ export default class SearchBar extends React.PureComponent {
...
-              style={[{
-                fontSize: 17,
-                color: this.props.tintColor || '#007AFF',
-              }, this.props.textFont ? { fontFamily: this.props.textFont } : null]}>
+              style={[
+                {
+                  fontSize: 17,
+                  color: this.props.tintColor || '#007AFF',
+                },
+                this.props.textFontFamily ? { fontFamily: this.props.textFontFamily } : null
```
Comment (eriveltonelias) [inline src/SearchBar.ios.js]: let's do this way:
```js
color: this.props.tintColor || '#007AFF',
...(this.props.textFontFamily &&  { fontFamily: this.props.textFontFamily })

```

**Review by eriveltonelias — CHANGES_REQUESTED** [review]
thanks for your PR! I just left a few comments.

File: src/SearchLayout.js (hunk 1 above)
Comment (summerkiflain) [reply src/SearchLayout.js:46]: done!

File: src/SearchBar.ios.js (hunk 2 above)
Comment (summerkiflain) [reply src/SearchBar.ios.js]: Thanks for the tip.

**Comment by summerkiflain** [comment]
@eriveltonelias PR updated, thanks for the feedback.

**Review by eriveltonelias — APPROVED** [review]
thanks!

**Comment by summerkiflain** [comment]
@eriveltonelias any chance you can publish this on npm?

**Pull request merged**